src/R/Components/Impl/Settings/IRSettings.cs

```diff
@@ -1,10 +1,12 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
+using System.ComponentModel;
+using Microsoft.Common.Core.Logging;
 using Microsoft.R.Components.ConnectionManager;
 
 namespace Microsoft.R.Components.Settings {
-    public interface IRSettings {
+    public interface IRSettings: INotifyPropertyChanged {
```

Comment:
This is not a view model, so I'd rather use our own event instead of `INotifyPropertyChanged`